### PR TITLE
Add license tests with new text #2300

### LINKED ISSUE #2300: Add license tests with new text
See https://github.com/Bacchustaf/SyRe_MSc/blob/078df81c8d13c873b15c92684167f4503bd766df/xfemm-code-310/mfemm/pfemm/fsolver/LICENCE.txt

### PR DESCRIPTION
**Pull request by sritasngh**
I have added Aladdin Free Public License, Other permissive license and proprietary license rule.

Signed-off-by: Sarita Singh <saritasingh.0425@gmail.com>
Fixes #2300 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by sritasngh** [comment]
Now I am getting this result: https://gist.github.com/itssingh/cdbd0899dee0d8ca31b8650d23680b8a. @pombredanne It seems to be correct. Does it require any other change?

File: src/licensedcode/data/rules/afpl-8.0_url_4.RULE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Please note that the program Triangle,
```
Comment (pombredanne) [inline src/licensedcode/data/rules/afpl-8.0_url_4.RULE]: Are you sure that a `_url` suffix in the name is needed?

File: src/licensedcode/data/rules/afpl-8.0_url_4.RULE [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Please note that the program Triangle, 
+written by Jonathan Shewchuk and used by FEMM
+for the purposes of mesh generation is NOT covered
+by this license and is subject to its own licensure terms.
```
Comment (pombredanne) [inline src/licensedcode/data/rules/afpl-8.0_url_4.RULE]: You should break this two smaller rules:

```
mesh generation is NOT covered by this license and is subject to its own  terms.
```
may be enough for one rule with a `minimum_coverage: 100`as this is rather unique. I have second thoughts about adding things like "licensure"  as each new added word has an impact with this suggestion there is no "licensre" anymore.

Also `distributed under its own licensing terms.  The licensing terms of Lua are also attached.` would then be  enough for an `other-permissive ` rule

File: src/licensedcode/data/rules/other-permissive_208.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+license_expression: other-permissive
+is_license_text: yes
+relevance: 99
```
Comment (pombredanne) [inline src/licensedcode/data/rules/other-permissive_208.yml]: Please add a `notes: ` field with some link to where such an old license can be found

File: src/licensedcode/data/rules/proprietary-license_518.RULE [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Created as part of the Archimedes project (tools for
```
Comment (pombredanne) [inline src/licensedcode/data/rules/proprietary-license_518.RULE]: `Created as part of the Archimedes project` may be enough for this rule with minimum_coverage: 100`

File: src/licensedcode/data/rules/proprietary-license_519.RULE [hunk 5]
```diff
@@ -0,0 +1,19 @@
+[Triangle] may be freely redistributed under the
```
Comment (pombredanne) [inline src/licensedcode/data/rules/proprietary-license_519.RULE]: We should remove the leading and trailing unique names as thy are making the index heavier but do not help with detection.

```suggestion
may be freely redistributed under the
```

File: src/licensedcode/data/rules/proprietary-license_519.RULE [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+UNDER COPYRIGHT OF THE ORIGINAL AUTHOR, BOTH SOURCE
+AND OBJECT CODE ARE MADE FREELY AVAILABLE WITHOUT
+CHARGE, AND CLEAR NOTICE IS GIVEN OF THE
+MODIFICATIONS.  Distribution of this code as part of
+a commercial system is permissible ONLY BY DIRECT
+ARRANGEMENT WITH THE AUTHOR.  (If you are not
+directly supplying this code to a customer, and you
+are instead telling them how they can obtain it for
+free, then you are not required to make any
+arrangement with [JRS].)
```
Comment (pombredanne) [inline src/licensedcode/data/rules/proprietary-license_519.RULE]:
```suggestion
arrangement with
```

File: src/licensedcode/data/rules/proprietary-license_519.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+license_expression: proprietary-license
+is_license_notice: yes
+relevance: 99
```
Comment (pombredanne) [inline src/licensedcode/data/rules/proprietary-license_519.yml]:
```suggestion
relevance: 100
```

File: src/licensedcode/data/rules/proprietary-license_518.yml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+license_expression: proprietary-license
+is_license_notice: yes
+relevance: 99
```
Comment (pombredanne) [inline src/licensedcode/data/rules/proprietary-license_518.yml]:
```suggestion
relevance: 100
```

File: src/licensedcode/data/rules/other-permissive_208.yml (hunk 3 above)
Comment (pombredanne) [inline src/licensedcode/data/rules/other-permissive_208.yml]:
```suggestion
relevance: 100
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you! Please see my comments for your consideration.

File: src/licensedcode/data/rules/afpl-8.0_url_4.RULE (hunk 1 above)
Comment (sritasngh) [reply src/licensedcode/data/rules/afpl-8.0_url_4.RULE]: No my mistake:( Changed it.

**Comment by sritasngh** [comment]
@pombredanne I have made all the changes. Please let me know if I missed something!

**Review by pombredanne — APPROVED** [review]
LGTM! Thank you ++ for pushing this through!

**Pull request merged**